virtualmachine/vsphere/util.go

```diff
@@ -522,29 +559,44 @@ func searchTree(vm *VM, mor *types.ManagedObjectReference, name string) (
 		name)
 }
 
-// searchVmByUuid: searches a datacenter for vm with uuid: instanceUuid
-func searchVmByUuid(vm *VM, instanceUuid string) (*mo.VirtualMachine, error) {
+// searchVmByUuid: searches vm with uuid: instanceUuid in datacenter
+// or entire inventory
+func searchVmByUuid(vm *VM, searchFilter VMSearchFilter) (
+	*mo.VirtualMachine, error) {
+	var (
+		dcObj *object.Datacenter
+		obj   object.Reference
+		err   error
+		dcMo  *mo.Datacenter
+	)
 	vmMo := mo.VirtualMachine{}
 	s := object.NewSearchIndex(vm.client.Client)
 	isInstanceUuid := new(bool)
 	*isInstanceUuid = true
-	dcMo, err := GetDatacenter(vm)
-	if err != nil {
-		return nil, err
+
+	if searchFilter.SearchInDC {
+		dcMo, err = GetDatacenter(vm)
+		if err != nil {
+			return nil, err
+		}
+		dcObj = object.NewDatacenter(vm.client.Client, dcMo.Self)
+		obj, err = s.FindByUuid(vm.ctx, dcObj, searchFilter.InstanceUuid, true,
```

Comment:
put it out side if else and set dcObj to nil

Reply:
dcObj is kept in if part because it is only used there and nowhere else. It is fine?